allure-support/src/main/kotlin/com/kaspersky/components/alluresupport/interceptors/testrun/MoveReportsInterceptor.kt

```diff
@@ -0,0 +1,133 @@
+package com.kaspersky.components.alluresupport.interceptors.testrun
+
+import android.app.Instrumentation
+import androidx.test.uiautomator.UiDevice
+import com.google.common.io.CharStreams
+import com.kaspersky.components.alluresupport.files.resources.AllureResourcesRootDirsProvider
+import com.kaspersky.kaspresso.files.dirs.DirsProvider
+import com.kaspersky.kaspresso.interceptors.watcher.testcase.TestRunWatcherInterceptor
+import com.kaspersky.kaspresso.testcases.models.info.TestInfo
+import io.qameta.allure.kotlin.Allure
+import io.qameta.allure.kotlin.util.PropertiesUtils
+import org.json.JSONObject
+import java.io.File
+
+private const val ATTACHMENTS_JSON_FIELD = "attachments"
+private const val NAME_JSON_FIELD = "name"
+private const val SOURCE_JSON_FIELD = "source"
+private const val MP4_EXTENSION = "mp4"
+
+/**
+ * Current allure version stores reports to /data/data/your.package.name/files/allure-results.
+ * This interceptor moves them to default artifacts folder after test and replaces mock videos with real ones
+ */
+class MoveReportsInterceptor(
+    private val instrumentation: Instrumentation,
+    private val dirsProvider: DirsProvider,
+    private val rootDirsProvider: AllureResourcesRootDirsProvider,
+    private val videosHolder: AttachedAllureVideosHolder,
+    private val device: UiDevice
+) : TestRunWatcherInterceptor {
+
+    private var lastTestCaseUuid: String? = null
+
+    override fun onMainSectionStarted(testInfo: TestInfo) {
+        lastTestCaseUuid = Allure.lifecycle.getCurrentTestCase()
+    }
+
+    override fun onTestFinished(testInfo: TestInfo, success: Boolean) {
+        val allureTargetDir = moveAllureReportToSdCard()
```

Comment:
If the user adds custom interceptors after this one he wouldn't be able to see its results in the report. For example `AllureSupportCustomizeTest` will right custom view hierarchy which will not be added to the report.

Reply:
First I would suggest to move here that hacky forcing the allure to write down the report to have something like:
```
    override fun onTestFinished(testInfo: TestInfo, success: Boolean) {
        forceAllureTestFinished(success)
        val allureTargetDir = moveAllureReportToSdCard()
        removeStubs(allureTargetDir)
        moveAttachedVideosToProperDirectories(allureTargetDir)
        cleanUp()
    }
    fun forceAllureTestFinished(success: Boolean) {
        val uuid = Allure.lifecycle.getCurrentTestCase() ?: “”
        Allure.lifecycle.updateTestCase {
            it.status = if (success) Status.PASSED else Status.FAILED
        }
        Allure.lifecycle.stopTestCase(uuid)
        Allure.lifecycle.writeTestCase(uuid)
    }
```
And then we should guarantee that this `MoveReportsInterceptor` is called last no matter how much interceptors the user added by himself. How to achieve this is need to be designed.